processing-and-analysis/02-command-files/Ganling's file/Random_effect_tables.Rmd

```diff
@@ -0,0 +1,122 @@
+View(data_2s_unique)
+data_2s_exam_score <- data_2s_unique %>% 
+  select(exam1, exam2, finalexam, course_fullid, ver, sex_id, urm_id, eop_id, fgn_id, experiment1, ta_sect, satverbal, satmath, high_sch_gpa, mastered_topics_initial_kc)
+  
+data_2s_exam_score_true <- na.omit(data_2s_exam_score)
+```
+##Make 'ta_sect'unique across different data values
+####First, we need to split the data from into year 16 and year 17 
+```{r}
+data_2s_exam_score_true$ta_sect <- as.character(data_2s_exam_score_true$ta_sect)
+
+#next, split the dataframe in half
+master_unique_2016 <- subset(data_2s_exam_score_true, course_fullid != "CHEM_142_A_2017_4" & course_fullid != "CHEM_142_B_2017_4")
+master_unique_2017 <- subset(data_2s_exam_score_true, course_fullid != "CHEM_142_A_2016_4" & course_fullid != "CHEM_142_B_2016_4")
+```
+####Use the vectorization function in R, we can add '_16' and '_17'at the end of each TA section 
+```{r}
+master_unique_2016$ta_sect <- paste0(master_unique_2016$ta_sect, "_16")
+master_unique_2017$ta_sect <- paste0(master_unique_2017$ta_sect, "_17")
+```
+##recombine the dataframes together and make the 'ta_sect' factors again
+```{r}
+master_true <- rbind(master_unique_2016, master_unique_2017)
+master_true$ta_sect <- as.factor(master_true$ta_sect)
+```
+####I would like to create a model that could implement the random effects based on the TA sections
+```{r}
+#First, include sex_id as a control variable
+mod1 <- lmer(finalexam ~ experiment1 + satmath + satverbal + mastered_topics_initial_kc + high_sch_gpa + sex_id + (1|ta_sect), 
+           data=master_true)
+summary(mod1)
+AIC(mod1)
+
+#Include eop_id as a control variable
+mod2 <- lmer(finalexam ~ experiment1 + satmath + satverbal + mastered_topics_initial_kc + high_sch_gpa + eop_id + (1|ta_sect), 
+           data=master_true)
+summary(mod2)
+AIC(mod2)
+
+#Include fgn_id as a control variable
+mod3 <- lmer(finalexam ~ experiment1 + satmath + satverbal + mastered_topics_initial_kc + high_sch_gpa + fgn_id + (1|ta_sect), 
+           data=master_true)
+summary(mod3)
+AIC(mod3)
+
+#Include urm_id as a control variable
+mod4 <- lmer(finalexam ~ experiment1 + satmath + satverbal + mastered_topics_initial_kc + high_sch_gpa + urm_id + (1|ta_sect), 
+           data=master_true)
+summary(mod4)
+AIC(mod4)
+#
+```
+####This is what I got previously by changing out different independent variables, now I wnat to compare only the models with and without random effects
+```{r}
+#First, test that the treatment impacts scores
+mod1 <- lm(finalexam ~ experiment1 + satmath + satverbal + mastered_topics_initial_kc + high_sch_gpa, 
+           data=master_true)
+summary(mod1)
+AIC(mod1)
+
+#Include random effects (clustering)
+mod2 <- lmer(finalexam ~ experiment1 + satmath + satverbal + mastered_topics_initial_kc + high_sch_gpa + (1|ta_sect), 
+           data=master_true)
+summary(mod2)
+AIC(mod2)
+```
+```{r}
+AIC(mod1, mod2)
+```
+####Again, the change of AIC is much bigger than 2
```

Comment:
You have to leave a space after the #### tag for markdown to interpret this line as a header.